pilot/pkg/bootstrap/server.go

```diff
@@ -320,6 +337,18 @@ func (s *Server) Start(stop <-chan struct{}) error {
 			return err
 		}
 	}
+	// Race condition - if waitForCache is too fast and we run this as a startup function,
+	// the grpc server would be started before CA is registered. Listening should be last.
```

Comment:
I'm confused. runca is a startup function and this is called immediately, so this runs before run ca? But the comment indicates it needs to run after

Reply:
It runs before, but starting the server is in another go-routine. So CA starts at the same time with 'wait for sync, start listening'. 

Start CA needs to register a grpc service. But if 'wait for sync' is too fast - like in case of local files, no k8s, or fast k8s - grpc will start listening before CA registered grpc service. And we fail without any log - since grpc does it's own logging that we don't configure. 

Spent few hours to get to it - it happens often with the 'local files' adapter.

Reply:
That makes sense, is the comment indicating that this is currently an issue, not one that is fixed in this pr?

Reply:
It is an existing behavior, this PR is fixing it.  Hard to repro in k8s - it's slow enough that this won't happen typically.